test-framework/common/src/main/java/io/quarkus/test/common/TestResourceManager.java

```diff
@@ -405,22 +432,20 @@ private Collection<AnnotationInstance> findQuarkusTestResourceInstances(Class<?>
             testClasses.add(current.getName());
             current = current.getEnclosingClass();
         }
-        Set<AnnotationInstance> testResourceAnnotations = new LinkedHashSet<>();
-        for (AnnotationInstance annotation : index.getAnnotations(DotName.createSimple(QuarkusTestResource.class.getName()))) {
-            if (keepTestResourceAnnotation(annotation, annotation.target().asClass(), testClasses)) {
-                testResourceAnnotations.add(annotation);
-            }
-        }
 
-        for (AnnotationInstance annotation : index
-                .getAnnotations(DotName.createSimple(QuarkusTestResource.List.class.getName()))) {
-            for (AnnotationInstance nestedAnnotation : annotation.value().asNestedArray()) {
-                // keep the list target
-                if (keepTestResourceAnnotation(nestedAnnotation, annotation.target().asClass(), testClasses)) {
-                    testResourceAnnotations.add(nestedAnnotation);
-                }
-            }
-        }
+        Set<AnnotationInstance> testResourceAnnotations = Stream.of(WITH_TEST_RESOURCE, QUARKUS_TEST_RESOURCE)
+                .flatMap(dotName -> index.getAnnotations(dotName).stream())
+                .filter(annotation -> keepTestResourceAnnotation(annotation, annotation.target().asClass(), testClasses))
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+
+        Stream.of(WithTestResource.List.class.getName(), QuarkusTestResource.List.class.getName())
+                .map(DotName::createSimple)
+                .flatMap(dotName -> index.getAnnotations(dotName).stream()
+                        .flatMap(annotation -> Arrays.stream(annotation.value().asNestedArray())
+                                .filter(nestedAnnotation -> keepTestResourceAnnotation(nestedAnnotation,
+                                        annotation.target().asClass(), testClasses))))
+                .forEach(testResourceAnnotations::add);
```

Comment:
This is just a `no` from me and you know why 😉

Reply:
I'll change it but I'll still argue it's way cleaner and easier to read than a bunch of nested for loops with if conditions.